packages/nx-sonarqube/src/executors/scan/utils/utils.ts

```diff
@@ -214,6 +217,50 @@ function combineOptions(
     ...envOptions.Options(),
   };
 }
+
+export async function projectPackageVersion(
```

Comment:
@albthali Should have caught this the first time, apologies.

Can you please simplify the code below as such:

1. Remove fs import. When creating generators and executors, its good practice to use as many of the devkit utilities provided by Nx.
2. Having one return statement is easier to read
3. Remove async/await code

Try something like this:

```
export function sonarVersion(
  context: ExecutorContext,
  options: ScanExecutorSchema
): string {
  let sonarVersion;

  if (options.projectVersion) {
    sonarVersion = options.projectVersion;
  } else {
    const projectName = context.projectName;
    const projectPackagePath = `${context.workspace.projects[projectName].root}/package.json`;
    
    if (fileExists(projectPackagePath)) {
      const projectPackageJson = readJsonFile<{ version: number }>(projectPackagePath)
      const projectPackageVersion = projectPackageJson.version;
      sonarVersion = projectPackageVersion.toString();
    } else if (fileExists('package.json')) {
      const workspacePackageJson = readJsonFile<{ version: number }>(`package.json`)
      const workspacePackageVersion = workspacePackageJson.version;
      sonarVersion = workspacePackageVersion.toString();
    } else {
      sonarVersion = '1.0.0';
    }
  }

  return sonarVersion;
}
```

Reply:
Removed "fs" and "fs/promise" and started using nx readJsonFile and readFileIfExisitng. Checked on e2e tests and they are working.

Reply:
@albthali I would simplify your branch to the above logic. There are some unnecessary checks and logging in your PR. Also, you have file conflicts which need resolution

Reply:
It is a little bit tricky since there are additional checks after reading the file. I need to ensure that the package version actually exists. That is why something like the following makes since 
`export function projectPackageVersion(
  context: ExecutorContext,
  options: ScanExecutorSchema
): string {
  const projectName = context.projectName;
  if (options.projectVersion) {
    return   options.projectVersion;
  }
  let version = getPackageJsonVersion(context.workspace.projects[projectName].root)
  if (version) {
    return version
  }
  version = getPackageJsonVersion()
  return version
}`
I believe it is a lot easier to read since the other way requires a lot of nesting

Reply:
@albthali multiple vs single return statement is defintely a matter of opinion. So long as you remove the extra debugging logs and clean up the unused code I will approve